components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -1534,4 +1501,214 @@ private boolean isDateTimeAndMSSql(String attributeName) throws OrganizationMana
+        } else {
+            return getOrgSqlStmtTailForLegacyRuntime(authorizedSubOrgsOnly);
+        }
+    }
+
+    private String getOrgSqlStmtTailForLegacyRuntime(boolean authorizedSubOrgsOnly)
+            throws OrganizationManagementServerException {
+
+        String getOrgSqlStmtTail = authorizedSubOrgsOnly ? GET_ORGANIZATIONS_TAIL_LEGACY
+                : GET_ORGANIZATIONS_TAIL_WITHOUT_PERMISSION_CHECK;
+
+        if (isOracleDB()) {
+            getOrgSqlStmtTail = authorizedSubOrgsOnly ? GET_ORGANIZATIONS_TAIL_ORACLE_LEGACY
+                    : GET_ORGANIZATIONS_TAIL_ORACLE_WITHOUT_PERMISSION_CHECK;
+        } else if (isMSSqlDB()) {
+            getOrgSqlStmtTail = authorizedSubOrgsOnly ? GET_ORGANIZATIONS_TAIL_MSSQL_LEGACY
+                    : GET_ORGANIZATIONS_TAIL_MSSQL_WITHOUT_PERMISSION_CHECK;
+        }
+        return getOrgSqlStmtTail;
+    }
+
+    private String getOrgSqlStmtTailForModernRuntime(boolean authorizedSubOrgsOnly, String applicationAudience)
+            throws OrganizationManagementServerException {
+
+        String getOrgSqlStmtTail = authorizedSubOrgsOnly
+                ? StringUtils.isNotBlank(applicationAudience)
+                ? GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS_TAIL
+                : GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS_TAIL
+                : GET_ORGANIZATIONS_TAIL;
+
+        if (isOracleDB()) {
+            getOrgSqlStmtTail = authorizedSubOrgsOnly
+                    ? StringUtils.isNotBlank(applicationAudience)
+                    ? GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS_TAIL_ORACLE
+                    : GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS_TAIL_ORACLE
+                    : GET_ORGANIZATIONS_TAIL_ORACLE;
+        } else if (isMSSqlDB()) {
+            getOrgSqlStmtTail = authorizedSubOrgsOnly
+                    ? StringUtils.isNotBlank(applicationAudience)
+                    ? GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS_TAIL_MSSQL
+                    : GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS_TAIL_MSSQL
+                    : GET_ORGANIZATIONS_TAIL_MSSQL;
+        }
+        return getOrgSqlStmtTail;
+    }
+
+    private String appendFilterQueries(String sqlStmt, FilterQueryBuilder filterQueryBuilder,
+                                       FilterQueryBuilder parentIdFilterQueryBuilder, String getOrgSqlStmtTail,
+                                       boolean recursive, String sortOrder) {
+
+        String parentIdFilterQuery = parentIdFilterQueryBuilder.getFilterQuery();
+        if (StringUtils.isBlank(parentIdFilterQuery)) {
+            return sqlStmt + filterQueryBuilder.getFilterQuery() +
+                    String.format(getOrgSqlStmtTail, SET_ID, recursive ? "> 0" : "= 1", sortOrder);
+        } else {
+            return sqlStmt + filterQueryBuilder.getFilterQuery() +
+                    String.format(getOrgSqlStmtTail, parentIdFilterQuery, recursive ? "> 0" : "= 1", sortOrder);
+        }
```

Comment:
```suggestion
        }
        return sqlStmt + filterQueryBuilder.getFilterQuery() +
                    String.format(getOrgSqlStmtTail, parentIdFilterQuery, recursive ? "> 0" : "= 1", sortOrder);
```
else block is not required becasue if condition is satisfied return something, otherwise always return the other thing

Reply:
This is now updated.